server/src/main/kotlin/org/elaastic/player/dashboard/DashboardModelFactory.kt

```diff
@@ -20,142 +20,184 @@
-            convertPhaseState(learnerStepsModel.evaluationState),
+            learnerStepsModel.responseSubmissionState.getDashboardState(),
+            learnerStepsModel.evaluationState.getDashboardState(),
             sequenceId = sequence.id
         )
 
+        val learners: MutableList<LearnerMonitoringModel> = getLearnerMonitoringModels(
+            sequenceMonitoringModel,
+            sequence,
+        )
+        sequenceMonitoringModel.setLearners(learners)
+
+        val previousSequence: Sequence? = sequenceService.findPreviousSequence(sequence)
+        val nextSequence: Sequence? = sequenceService.findNextSequence(sequence)
+
+        return DashboardModel(
+            sequence,
+            previousSequence?.id,
+            nextSequence?.id,
+            sequenceMonitoringModel
+        )
+    }
+
+    private fun getLearnerMonitoringModels(
+        sequenceMonitoringModel: SequenceMonitoringModel,
+        sequence: Sequence,
+    ): MutableList<LearnerMonitoringModel> {
+        val attendees: List<LearnerAssignment> = assignmentService.getRegisteredUsers(sequence.assignment!!)
+        val responses: List<Response> = interactionService.findAllResponsesBySequenceOrderById(sequence)
+
+        // Associate each learner with the number of evaluations he made
+        val evaluationCountByUser = peerGradingService.countEvaluationsMadeByUsers(attendees, sequence)
+        val countResponseGradable = getCountResponseGradable(sequence)
+
         val learners: MutableList<LearnerMonitoringModel> = mutableListOf()
 
         attendees.forEach {
             learners.add(
-                LearnerMonitoringModel(
-                    it.learner.id!!,
-                    it.learner.getFullname(),
-                    getAttendeeStateOnResponsePhase(it, responses, sequenceMonitoringModel.phase1State),
-                    getAttendeeStateOnEvaluationPhase(
-                        it,
-                        sequence,
-                        sequenceMonitoringModel.phase2State,
-                        evaluationCountByUser,
-                        learnerToIfTheyAnswer,
-                        countResponseGradable,
-                    ),
-                    sequenceMonitoringModel = sequenceMonitoringModel
+                buildLearnerMonitoringModel(
+                    it,
+                    learnerHasAnswer(responses, it),
+                    sequenceMonitoringModel,
+                    sequence,
+                    evaluationCountByUser[it] ?: 0,
+                    countResponseGradable
                 )
             )
         }
+        return learners
+    }
 
-        sequenceMonitoringModel.setLearners(learners)
+    /**
+     * Get the number of responses that are gradable.
+     *
+     * @param sequence the sequence
+     * @return the number of responses that are gradable
+     */
+    fun getCountResponseGradable(sequence: Sequence): Long {
```

Comment:
Minor remark: it would be more common to name this function `countXXX` than `getCountXXX`.

Funny thing: I would have named `GradableResponse` instead of `ResponseGradable`.
Well... I'm not totally sure of that one.
Here are the arguments I can think about : 
- It's more english compliant (adjective before the noun)
- For class name, it is quite conventional to this naming strategy : `<Implementation><Concept><SubConcept...>`. e.g. the interface `Session` and its implementations `DefaultSession`, `LocalThreadSession` ...
Here "gradable" is not a subconcept ; I would see it as a qualifier of the concept "response".